### DISCUSSION TITLE
How do I prevent DTEND being day+1

### DISCUSSION THREAD

**Discussion by noeldum** [markuspoerschke/iCal#255] [Q&A] [answered]
How do I prevent DTEND being day+1 in the output?

I tried both 
$vEvent->setDtEnd(new \DateTimeImmutable('2012-12-24'));
and 
$vEvent->setDtEnd(new \DateTime('2012-12-24'));

in both cases I get 
DTEND;VALUE=DATE:20121225

instead of 20121224

**Comment by markuspoerschke** [comment, marked as answer]
I assume, that you want to create a single day event that takes place on December 24th 2012.

The same behaviour can be examined when using the [example1.php](https://github.com/markuspoerschke/iCal/blob/master/examples/example1.php).

I created the following iCalendar file using the following code:

```php
$vEvent = new \Eluceo\iCal\Component\Event();
$vEvent->setDtStart(new \DateTime('2020-12-24'));
$vEvent->setDtEnd(new \DateTime('2020-12-24'));
$vEvent->setNoTime(true);
$vEvent->setSummary('Christmas');

$vCalendar = new \Eluceo\iCal\Component\Calendar('www.example.com');
$vCalendar->addComponent($vEvent);

echo $vCalendar->render();
```

The output will contain the following lines:

```ics
DTSTART;VALUE=DATE:20201224
DTEND;VALUE=DATE:20201225
```

At first glance, it looks like an event that occurs on the 24th and 25th is created.

Importing the [iCalendar file](https://github.com/markuspoerschke/iCal/files/4659347/issue-160.ics.txt) into Windows' calendar app will show the following result:

![grafik](https://user-images.githubusercontent.com/1222377/82498491-5fe6a100-9af0-11ea-9bf5-c61c0b3f5c2c.png)

The event is created as a single day event on the 24th only.

If we check the [specification](https://tools.ietf.org/html/rfc5545#page-54), we will see the reason:

> The "DTSTART" property for a "VEVENT" specifies the inclusive start of the event. [...]
> The "DTEND" property for a "VEVENT" calendar component specifies the non-inclusive end of the event. [...]

In other words, the event will take place for the following interval:

```
DATE >= 2020-12-24
AND
DATE < 2020-12-25
````
This is why `DTEND` represents the first day on which the event no longer takes place.

**Comment by nekromoff** [comment]
Came here for the same thing, thanks for the explanation.

**Comment by lmollard** [comment]
Hi,

I have the same problem, but with an event more than 1 day long. The end date is the day I sent to the $vEvent->setDtEnd function + 1 day.

The resulting ICS file have the end date i want +1.

When i add this ICS file to Google Calendar, it works well (with the end date I need), but with the same ICS file, when I use it on Abritel and Airbnb, it's the end date +1 day.

How can I do for my ICS file to work with Abritel and Airbnb correctly ?

Thank you for your help.

Best regards

**Comment by markuspoerschke** [comment]
@lmollard 

> How can I do for my ICS file to work with Abritel and Airbnb correctly ?

I cannot give support for special consumers of the resulting ics file.

Maybe, if you provide an example PHP code and the resulting ICS file can help to see the problem.

**Comment by lmollard** [comment]
Hi, 

Thank you for your reply... I managed to get my ICS file to work as expected by setting the beggining and ending hour, so my code changed since my original demand.

Thank you again for trying to help me.

Lionel
